lib/stripe.js

```diff
@@ -296,21 +296,29 @@ Stripe.prototype = {
   getClientUserAgentSeeded: function(seed, cb) {
     var self = this;
 
-    exec('uname -a', function(err, uname) {
-      var userAgent = {};
-      for (var field in seed) {
-        userAgent[field] = encodeURIComponent(seed[field]);
-      }
+    try {
+      exec('uname -a', function(err, uname) {
```

Comment:
@rattrayalex-stripe Should we make an effort to handle this `err`? Unless I missed something in the docs, it doesn't seem like calling `exec` is guaranteed to always throw:

> If a callback function is provided, it is called with the arguments (error, stdout, stderr). On success, error will be null. On error, error will be an instance of Error. The error.code property will be the exit code of the child process while error.signal will be set to the signal that terminated the process. Any exit code other than 0 is considered to be an error.

Reply:
Good catch. I'll take a look at this and follow up.

Reply:
That's a good question - I left it as is to not change the API contract. 

From what I could tell, the original design goal was to let L316 optimistically try to encode whatever was passed in and fail to "UNKNOWN". 

Per https://github.com/nodejs/node/blob/908292cf1f551c614a733d858528ffb13fb3a524/lib/internal/child_process.js#L350-L354,`exec` looks like it will only throw if the system error is _not_:

```
  if (err === UV_EACCES ||
      err === UV_EAGAIN ||
      err === UV_EMFILE ||
      err === UV_ENFILE ||
      err === UV_ENOENT) {
```
Otherwise, any other system errors will throw an error. So it's not exactly consistent as to what will throw.

Reply:
@brennentsmith Great, that's what I was hoping!

Would you mind writing two tests to demonstrate this?

1. mock `exec` to throw an error, observe that UA is `"UNKNOWN"`
2. mock `exec` to provide an `err` variable to the callback, observe that UA is `"UNKNOWN"`

No worries if not – I can probably take.

Reply:
@brennentsmith Yeah, I could see from the diff that it seems like we've been ignoring the error return fro a while :)

But yeah, IMO since we're tightening up this code here anyway (and you've introduced a convenient helper to work with), we should just try to handle either class of error that might come back.

Reply:
I'll take this 😄 thanks for your help @brennentsmith !

Reply:
https://github.com/stripe/stripe-node/pull/598

(FTR, this PR handled the "error callback" case as well as the original code did)